0.19 support

lukewestby opened the pull request:
This renames all references to `elm-make` to `elm make` and adds the `--optimize` flag. Since 0.18's `elm` binary will run `elm-make` when `elm make` is invoked this upgrade _should_ work for both 0.18 and 0.19.

stoeffel commented on src/CliArguments.hs:
```diff
@@ -39,10 +40,12 @@ parser = do
   debug <-
     switch (long "debug" <> short 'd' <> help "Run jetpack in debug mode.")
-  warn <- switch (long "warn" <> short 'w' <> help "Output elm make warnings.")
+  warn <-
```
should we drop warn?

stoeffel replied:
It’s also not set anymore in compiler

stoeffel replied:
https://github.com/NoRedInk/jetpack/search?q=warn&unscoped_q=warn

stoeffel commented on src/CliArguments.hs:
```diff
@@ -60,6 +63,10 @@ parser = do
           then True
           else warn
+    , optimize =
+        if optimize
```
would be nice to use a custom type for this. I should have done that for the other flags as well.

lukewestby commented:
@stoeffel I removed warnings and made the compile mode a union type. I also modified the npm package locally to point to my build and ran it on the monolith and it looks like the progress bar is correct and it fails correctly.

lukewestby commented:
@stoeffel ready for a publish whenever you are!
